vpr/src/place/move_generator.h

```diff
@@ -23,14 +23,19 @@ struct MoveOutcomeStats {
 /**
  * @brief A Struct to hold statistics about the different move types
  *
- * num_moves:      save the number of proposed moves of each type (e.g. indexed from 0 to NUM_PL_MOVE_TYPES-1 )
- * accepted_moves: save the number of accepted moves of each type (e.g. indexed from 0 to NUM_PL_MOVE_TYPES-1 )
- * aborted_moves:  save the number of aborted moves of each type (e.g. indexed from 0 to NUM_PL_MOVE_TYPES-1 )
+ * num_moves:      save the number of proposed moves of each move type(e.g. [0..NUM_PL_MOVE_TYPES-1])
```

Comment:
I don't think you need to say "save" over and over again.  Just say 
"number of proposed moves of each move type (e.g. ..."

Reply:
1. As much as possible, get rid of some of this state, and just store everything over agent_available_types, and then post-process.  Less state to describe here, and less state to update in a cpu critical routine.

2. Longer term: explore having the 2nd state also have (block_type,move_type) but critical random and feasible region are special and don't pay attention to the block type. 

3. (If not removed) Rename num_moves: moves_per_move_type and aborted_moves.

Reply:
Possibly lets you make a vector of a single structure (might be slightly easier to read).